tests/components/solaredge/test_coordinator.py

```diff
@@ -0,0 +1,77 @@
+"""Tests for the SolarEdge coordinator services."""
+from unittest.mock import Mock, patch
+
+import pytest
+
+from homeassistant.components.solaredge.coordinator import SolarEdgeOverviewDataService
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+SITE_ID = "1a2b3c4d5e6f7g8h"
+
+mock_overview_data = {
+    "overview": {
+        "lifeTimeData": {"energy": 100000},
+        "lastYearData": {"energy": 50000},
+        "lastMonthData": {"energy": 10000},
+        "lastDayData": {"energy": 0.0},
+        "currentPower": {"power": 0.0},
+    }
+}
+
+
+@patch("solaredge.Solaredge")
+def test_solaredgeoverviewdataservice_valid_energy_values(mock_solaredge):
+    """Test valid no exception for valid overview data."""
+    data_service = SolarEdgeOverviewDataService(Mock(), mock_solaredge, SITE_ID)
```

Comment:
Please test by setting up Home Assistant and running it from our core API instead of testing the specific service as done here.

Reply:
I've changed the test to use the core API instead. Hope it's good enough. :)